### PR TITLE
Fix bug in BolusEntryView with "save without bolusing"

### PR DESCRIPTION
**Pull request by rickpasetto**
This is a regression introduced by my recent changes (thanks, Darin, for finding it!)

To make this a bit easier to reason about, I've moved a bunch of "helpers" into the ViewModel where it could be more easily unit tested.

### CODE REVIEW

File: Loop/View Models/BolusEntryViewModel.swift [hunk 1]
```diff
@@ -601,3 +601,51 @@ final class BolusEntryViewModel: ObservableObject {
...
+            return false
+        } else if activeNotice != .staleGlucoseData {
+            return true
+        } else {
+            return !isManualGlucoseEntryEnabled
+        }
+    }
+    
+    private var hasBolusEntryReadyToDeliver: Bool {
+        return enteredBolus.doubleValue(for: .internationalUnit()) != 0
```
Comment (darinkrauss) [inline Loop/View Models/BolusEntryViewModel.swift]: [nit] Consistent use of `return` in these one-liners?

File: Loop/View Models/BolusEntryViewModel.swift [hunk 2]
```diff
@@ -601,3 +601,51 @@ final class BolusEntryViewModel: ObservableObject {
 extension BolusEntryViewModel.Alert: Identifiable {
     var id: Self { self }
 }
+
+// MARK: Helpers
+extension BolusEntryViewModel {
+    var isManualGlucosePromptVisible: Bool {
+        activeNotice == .staleGlucoseData && !isManualGlucoseEntryEnabled
+    }
+    var isNoticeVisible: Bool {
```
Comment (darinkrauss) [inline Loop/View Models/BolusEntryViewModel.swift:612]: [nit] Missing blank line?

**Review by darinkrauss — APPROVED** [review]
A couple nits. I *think* it looks right, but I haven't been involved in this code, so I'll defer to @novalegra as to its correctness. 🚓

**Comment by novalegra** [comment]
What _is_ the "save without bolusing" bug? I've played around with the flow in the current build and am unsure as to what it is

**Comment by darinkrauss** [comment]
@novalegra Here is what I did... 

Tap Carb button on main window. Enter 1 Carb, tap Continue.  Tap Save without Bolusing. The button goes away, but the dialog still shows. 

From reading the current code (prior to this PR), it looks like it will only save if there is a non-zero bolus:
```
        Button(
            action: {
                if !self.hasBolusEntryReadyToDeliver {
                    self.shouldBolusEntryBecomeFirstResponder = true
                } else {
                    self.viewModel.saveAndDeliver(onSuccess: self.dismiss)
                }
            },
```
and:
```   
    private var hasBolusEntryReadyToDeliver: Bool {
        return self.viewModel.enteredBolus.doubleValue(for: .internationalUnit()) != 0
    }
```

**Review by novalegra — APPROVED** [review]
This LGTM!

**Pull request merged**